### PR TITLE
:wrench: [3.x] Make the sameple runnable and cleanup the gradle a bit

### PR DESCRIPTION
**Pull request by kittinunf**
### What's in this PR?

- Make the project compilable under Explicit API mode
- Make the application runnable using `gradle` application plugin
- Improve interface for moshi just a bit

### CODE REVIEW

File: fuel-base/src/test/kotlin/fuel/HttpLoaderBuilderTest.kt [hunk 1]
```diff
@@ -12,7 +12,8 @@ import org.junit.Test
 import java.net.SocketTimeoutException
 import java.util.concurrent.TimeUnit
 
-class HttpLoaderBuilderTest {
+internal class HttpLoaderBuilderTest {
```
Comment (kittinunf) [inline fuel-base/src/test/kotlin/fuel/HttpLoaderBuilderTest.kt:15]: Without a specific accessible modifier for explicit API mode, Kotlin compiler doesn't allow it.

File: build.gradle.kts [hunk 2]
```diff
@@ -27,7 +27,9 @@ allprojects {
     }
 
     val isSample = project.name in sampleModules
-    if (!isSample) {
+    if (isSample) {
+        apply(plugin = "org.gradle.application")
```
Comment (kittinunf) [inline build.gradle.kts:31]: This is to make our sample runnable under `gradlew run`

File: fuel-samples/weather/src/main/kotlin/fuel/samples/FuelWeather.kt [hunk 3]
```diff
@@ -6,19 +6,41 @@ import fuel.Fuel
...
+)
+
+@JsonClass(generateAdapter = true)
+data class ConsolidatedWeatherEntry(
+        val applicable_date: String,
+        val weather_state_name: String,
+        val the_temp: Float
 )
 
 fun main() {
```
Comment (kittinunf) [inline fuel-samples/weather/src/main/kotlin/fuel/samples/FuelWeather.kt:30]: This produces somewhat more meaningful command-line application :p

Screen Shot 2021-02-22 at 15 40 11

File: fuel-moshi/src/test/kotlin/fuel/moshi/FuelMoshiTest.kt [hunk 4]
```diff
@@ -39,6 +39,25 @@ internal class FuelMoshiTest {
...
+    fun testReifiedTypeMoshiResponse() = runBlocking {
+        val mockWebServer = MockWebServer().apply {
+            enqueue(MockResponse().setBody("{\"userAgent\": \"Fuel\"}"))
+        }
+
+        withContext(Dispatchers.IO) {
+            mockWebServer.start()
+        }
+
+        val response = Fuel.get(mockWebServer.url("user-agent"))
```
Comment (iNoles) [inline fuel-moshi/src/test/kotlin/fuel/moshi/FuelMoshiTest.kt:52]: jscpd found too many duplicates (7.69%) over threshold (0%)

Change these variables with different name

Comment (kittinunf) [reply fuel-moshi/src/test/kotlin/fuel/moshi/FuelMoshiTest.kt:52]: What is jscpd? ... 😅

Comment (kittinunf) [reply fuel-moshi/src/test/kotlin/fuel/moshi/FuelMoshiTest.kt:40]: Okay fixed, maybe.

**Review by iNoles — APPROVED** [review]
It does good now.

**Comment by kittinunf** [comment]
Thanksss

**Pull request merged**